src/ngRoute/route.js

```diff
@@ -786,7 +788,10 @@ function $RouteProvider() {
         if (angular.isDefined(template)) {
           locals['$template'] = template;
         }
-        return $q.all(locals);
+        return $q.all(locals).then(function(locals) {
+          $browser.$$completeOutstandingRequest(function() {});
+          return locals;
+        });
```

Comment:
This should be `$.all(locals).finally(function() { $browser.$$completeOutstandingRequest(function() {}); });`.
Because now you will never reach 0 outstanding requests once a resolve fails in the app.